src/eos_gates.js

```diff
@@ -101,13 +105,19 @@ const Application = new Lang.Class({
     },
 
     getHelpMessage: function() {
-        if (this._alreadyHaveReplacement)
-            return _("However, you already have <b>%s</b> installed on this Computer").format(this.replacement.appName);
+        if (!this.replacement)
+            return _("You can install applications from our <a href='endlessm-app://eos-app-store'>App Store</a>.");
+
+        let isEquivalentApp = !!this.replacement.replacementInfo;
+        let appName = !!this.replacement.replacementInfo ?
+                      this.replacement.replacementInfo.appName : this.replacement.appName;
+        let additionalInfo = !!this.replacement.replacementInfo ?
+                      ', ' + this.replacement.replacementInfo.description + ',' : '';
```

Comment:
This really won't work for translations; you can't just concatenate a comma like that and expect that it will look right in all languages.
To be honest, I would remove the concept of additional description entirely.